### DISCUSSION TITLE
Add conversion to DateOnly from DateTime and DateTimeOffset

### DISCUSSION THREAD

**Discussion by mkorsukov** [dotnet/runtime#103440] [Ideas] [answered]
Hi .NET team and community!

I found the `DateOnly` type is so useful in daily use cases, so I think about an addition to `DateTime` and `DateTimeOffset` types in the form of `ToDateOnly()` instance method.

Here is code snippets:

[DateTime.cs](/runtime/src/libraries/System.Private.CoreLib/src/System/DateTime.cs)

```cs
public DateOnly ToDateOnly() =>
    DateOnly.FromDateTime(this);
```

[DateTimeOffset.cs](/runtime/src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.cs)

```cs
public DateOnly ToDateOnly() =>
    DateOnly.FromDateTime(_dateTime);
```

Can't create a PR, so unit tests included.

[DateTimeTests.cs](/runtime/src/libraries/System.Runtime/tests/System.Runtime.Tests/System/DateTimeTests.cs)

```cs
[Fact]
public static void ToDateOnly()
{
    DateTime dateTime = new DateTime(2024, 06, 13, 11, 55, 45, 0, DateTimeKind.Utc);
    Assert.Equal(new DateOnly(2024, 06, 13), dateTime.ToDateOnly());
}
```

[DateTimeOffsetTests.cs](/runtime/src/libraries/System.Runtime/tests/System.Runtime.Tests/System/DateTimeOffsetTests.cs)

```cs
[Fact]
public static void ToDateOnly()
{
    DateTimeOffset dateTimeOffset = new DateTimeOffset(new DateTime(2024, 06, 13, 11, 55, 45, 0, DateTimeKind.Utc));
    Assert.Equal(new DateOnly(2024, 06, 13), dateTimeOffset.ToDateOnly());
}
```

**Comment by colejohnson66** [comment]
API requests go in the issues section

**Comment by huoyaoyuan** [comment, marked as answer]
Duplicate of #58734
